flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/api/TableEnvironmentITCase.scala

```diff
@@ -447,6 +447,50 @@ class TableEnvironmentITCase(tableEnvName: String, isStreaming: Boolean) extends
     assertFirstValues(sinkPath)
   }
 
+  @Test
+  def testTableDMLSync(): Unit = {
+    tEnv.getConfig.getConfiguration.set(TableConfigOptions.TABLE_DML_SYNC, Boolean.box(true));
+    val sink1Path = _tempFolder.newFolder().toString
+    tEnv.executeSql(
+      s"""
+         |create table MySink1 (
+         |  first string
+         |) with (
+         |  'connector' = 'filesystem',
+         |  'path' = '$sink1Path',
+         |  'format' = 'testcsv'
+         |)
+       """.stripMargin
+    )
+
+    val sink2Path = _tempFolder.newFolder().toString
+    tEnv.executeSql(
+      s"""
+         |create table MySink2 (
+         |  first string
+         |) with (
+         |  'connector' = 'filesystem',
+         |  'path' = '$sink2Path',
+         |  'format' = 'testcsv'
+         |)
+       """.stripMargin
+    )
+
+    val tableResult1 =
+      tEnv.sqlQuery("select first from MyTable").executeInsert("MySink1", false)
+    checkInsertTableResult(tableResult1, "default_catalog.default_database.MySink1")
+    assertFirstValues(sink1Path)
+
+    val tableResult2 =
+      tEnv.sqlQuery("select first from MySink1").executeInsert("MySink2", false)
+    checkInsertTableResult(tableResult2, "default_catalog.default_database.MySink2")
```

Comment:
`checkInsertTableResult` will wait until job is finished, we have to assert file data first and then check the `TableResult`.

Reply:
This is not resolved.